pkg/build/controller/factory/factory.go

```diff
@@ -14,16 +14,17 @@ import (
 	"github.com/GoogleCloudPlatform/kubernetes/pkg/watch"
 
 	buildapi "github.com/openshift/origin/pkg/build/api"
+	buildclient "github.com/openshift/origin/pkg/build/client"
 	controller "github.com/openshift/origin/pkg/build/controller"
 	strategy "github.com/openshift/origin/pkg/build/controller/strategy"
-	buildutil "github.com/openshift/origin/pkg/build/util"
 	osclient "github.com/openshift/origin/pkg/client"
 	imageapi "github.com/openshift/origin/pkg/image/api"
 )
 
 type BuildControllerFactory struct {
-	Client              *osclient.Client
-	KubeClient          *kclient.Client
+	OSClient            osclient.Interface
+	KubeClient          kclient.Interface
+	BuildUpdater        buildclient.BuildUpdateInterface
```

Comment:
i find it helpful when things are clearly spelled out as being interfaces w/o having to dig into them.  it tells me immediately that the implementation might be flexible and i need to consider why and how it changes, and tells me i can potentially change it if needed.  Having to jump to the definition to determine that is an extra step.

but again, i'll change it, i'm not that hung up on it, but i do think there are multiple valid schools of thought here.